pkg/apis/marketplace/v1alpha1/catalogsourceconfig_types.go

```diff
@@ -129,5 +134,19 @@ func (csc *CatalogSourceConfig) RemoveOwner(ownerUID types.UID) {
 
 // GetPackageIDs returns the list of package(s) specified.
 func (spec *CatalogSourceConfigSpec) GetPackageIDs() []string {
-	return strings.Split(strings.Replace(spec.Packages, " ", "", -1), ",")
+	return getValidPackageSliceFromString(spec.Packages)
+}
+
+func getValidPackageSliceFromString(pkgs string) []string {
+	pkgIds := make([]string, 0)
+
+	pkgSlice := strings.Split(strings.Replace(pkgs, " ", "", -1), ",")
+
+	for _, pkg := range pkgSlice {
```

Comment:
Can't you just trim the string instead of doing this? This can be done right in `GetPackageIDs()`. Or is there another reason for this?

Reply:
@aravindhp This solves the case not just for leading commas, but for any item in the list that isn't valid. Ex:

`etcd,,prometheus`
`,etcd,prometheus,`
`,etcd,,,prometheus,`

etc.

Ultimately this fix ensures that, since we are `split`ting on `","` we should ensure that we don't include any accidental additional elements.

This also means that, in the future, we could put further validation in this one method -- for example, if we want to restrict other special characters or key words or something else.